### ISSUE TITLE
Support sonatype nexus as registry provider

### ISSUE THREAD

**Issue by JoschuaA4** [goharbor/harbor#15422]
It would be usefull to have support for sonatype Nexus as a resgitry provider.

Actually i'am not able to support the community with code but hopefully i can test proposed implementaion.

Versions:
Please specify the versions of following systems.

- harbor version: [Version v2.2.2-56d7937f]

**Comment by phin1x** [comment]
@JoschuaA4 did you try the "docker registry" provider?

**Comment by wy65701436** [comment]
yes, if the sonatype Nexus is the OCI compitable registry, you can try docker registry first.

**Comment by JoschuaA4** [comment]
Sorry for the late reponse.
I tried to add the nexus registry via the "docker registry" provider.
But the UI repsonses with "Failed to ping endpoint." if i rune the connection test and with "the registry is unhealthy" if i click OK
The "Docker Hub" Provider does not behave like this?
I tried to add the fqdn from our private registry with 127.0.0.1 as described in [#5747](https://github.com/goharbor/harbor/issues/5747) but with any positive result.

**Comment by BulatSaif** [comment]
I have the same issue with old version v1, I hope these logs will help to solve the problem:
```
harbor-core      | 2021-08-24T13:02:15Z [WARNING] [/common/utils/registry/auth/tokenauthorizer.go:266]: Schemas [{basic map[realm:Sonatype Nexus Repository Manager]}] are unsupported
harbor-core      | 2021-08-24T13:02:15Z [WARNING] [/common/utils/registry/auth/tokenauthorizer.go:315]: empty realm, skip
harbor-core      | 2021-08-24T13:02:15Z [ERROR] [/replication/adapter/native/adapter.go:154]: failed to ping registry https://nexus.*****: http error: code 401, message 
```

**Comment by bitsf** [comment]
did you `docker login nexus.***** -u user -p pass` successfully ?

**Comment by Lyrwon** [comment]
Hi i have the same issue.

I have two servers in different locations / networks.
One for Nexus, the other for Harbor.

On the Harbor-Server i can connect with `docker login nexus.abc.def -u user -p pass`
In the Harbor UI get this errormessage in the harbor-core container logs:
`[ERROR] [/pkg/reg/adapter/native/adapter.go:126]: failed to ping registry http://XXX.XXX.XXX.XXX:8082: Get "http://XXX.XXX.XXX.XXX:8082/v2/": dial tcp XXX.XXX.XXX.XXX:8082: connect: connection refused`

Why do i need to `ping` my nexus?
___________________________________________

I had no issues while i tried the setup on a VMBox on my windows, there i started both services on the same server.

**Comment by bitsf** [comment]
what is port `8082` for ?
`ping` is just to connect the registry v2 api to verify the connectivity

**Comment by Lyrwon** [comment]
Hi, 

at the moment i'm using Version v2.3.4-7a068b12.

I used tcpdump to whatch on the packages while i tried to connect from Harbor to nexus. But there weren't any packages.
After this I commented out all the lines with **dns_search: .** in the docker-compose.yml an restarted all Containers.
Now it works.

What effekt should this lines have?

**Comment by EugenMayer** [comment]
We have the same issue.

- Using docker-registry as type
- Using our usual https/LE signed domain for our registry (sonatype)
- using the regular username as client id and password access secret 

- When using the button  test, it errors with 'Failed to ping endpoint'
- When pressing ok, it errors  `Failed to ping endpoint.`

This happens with v1 support enabled or disabled (on sonatype nexus). We use the latest sonatype nexus release.

**Comment by balnbibarbi** [comment]
We just tried this, and it seems to work fine. Maybe Nexus should be added to the list of supported third-party solutions?

In Nexus OSS 3.29.2-02 we uploaded an image. In Harbor, we created an endpoint of type "Docker Registry" pointing to the Nexus URL, with valid credentials for it, then created a "proxy cache" project. A pull of the image from the empty Harbor project succeeds, and the image is cached in it, as expected.

**Comment by balnbibarbi** [comment]
It's not entirely obvious that one should choose "Docker Registry" when attempting to interoperate with Nexus though. Perhaps references to "Docker Registry" everywhere could be renamed to "Generic OCI compliant registry"?
Or perhaps add a copy of "Docker Registry" under the name "Generic OCI compliant registry" or "Sonatype Nexus" in the documentation and UI?
